llama_stack/providers/adapters/inference/bedrock/bedrock.py

```diff
@@ -0,0 +1,359 @@
+                required.append(name)
+            json_properties[name] = json_property
+
+        input_schema['properties'] = json_properties
+        if required:
+            input_schema['required'] = required
+        return input_schema
+
+    @staticmethod
+    def _tools_to_tool_config(tools: Optional[List[ToolDefinition]], tool_choice: Optional[ToolChoice]) -> (
+            Optional)[Dict]:
+        if not tools:
+            return None
+
+        bedrock_tools = []
+        for tool in tools:
+            tool_name = tool.tool_name if isinstance(tool.tool_name, str) else tool.tool_name.value
+
+            tool_spec = {
+                "toolSpec": {
+                    "name": tool_name,
+                    "inputSchema": {
+                        "json": BedrockInferenceAdapter._tool_parameters_to_input_schema(tool.parameters),
+                    }
+                }
+            }
+
+            if tool.description:
+                tool_spec['toolSpec']['description'] = tool.description
+
+            bedrock_tools.append(tool_spec)
+        tool_config = {
+            "tools": bedrock_tools,
+        }
+
+        if tool_choice:
+            tool_config["toolChoice"] = {"any": {}} if tool_choice.value == ToolChoice.required else {"auto": {}}
+        return tool_config
+
+    async def chat_completion(
+            self,
+            model: str,
+            messages: List[Message],
+            sampling_params: Optional[SamplingParams] = SamplingParams(),
+            # zero-shot tool definitions as input to the model
+            tools: Optional[List[ToolDefinition]] = [],
+            tool_choice: Optional[ToolChoice] = ToolChoice.auto,
+            tool_prompt_format: Optional[ToolPromptFormat] = ToolPromptFormat.json,
+            stream: Optional[bool] = False,
+            logprobs: Optional[LogProbConfig] = None,
+    ) -> AsyncGenerator:  #Union[ChatCompletionResponse, ChatCompletionResponseStreamChunk]:
+        bedrock_model = BedrockInferenceAdapter.resolve_bedrock_model(model)
+        inference_config = BedrockInferenceAdapter.get_bedrock_inference_config(sampling_params)
+
+        tool_config = BedrockInferenceAdapter._tools_to_tool_config(tools, tool_choice)
+        bedrock_messages, system_bedrock_messages = self._messages_to_bedrock_messages(messages)
+
+        converse_api_params = {
+            "modelId": bedrock_model,
+            "messages": bedrock_messages,
+        }
+        if inference_config:
+            converse_api_params["inferenceConfig"] = inference_config
+
+        # Tool use is not supported in streaming mode
+        if tool_config and not stream:
+            converse_api_params["toolConfig"] = tool_config
+        if system_bedrock_messages:
+            converse_api_params["system"] = system_bedrock_messages
+
+        if not stream:
+            converse_api_res = self.client.converse(**converse_api_params)
+            print(converse_api_res)

```

Comment:
delete?

Reply:
Yes I used this for debugging accidentally left this in. Will remove.